src/bundle/Controller/UniversalDiscoveryController.php

```diff
@@ -161,58 +153,67 @@ public function accordionGridViewAction(
         Request $request,
         int $locationId
     ): JsonResponse {
-        $location = $this->loadLocation($locationId);
-
         $limit = $request->query->getInt('limit', 25);
         $sortClauseName = $request->query->getAlpha('sortClause', self::SORT_CLAUSE_DATE_PUBLISHED);
         $sortOrder = $request->query->getAlpha('sortOrder', Query::SORT_ASC);
+        $rootLocationId = $request->query->getInt('rootLocationId', self::ROOT_LOCATION_ID);
 
         $sortClause = $this->getSortClause($sortClauseName, $sortOrder);
 
-        $columns = $this->getColumns($location, $limit, $sortClause, true);
+        $columns = $this->getColumns($locationId, $limit, $sortClause, true, $rootLocationId);
 
         return new JsonResponse([
             'breadcrumb' => array_map(
                 function (Location $location) {
                     return $this->getRestFormat($location);
                 },
-                $this->getBreadcrumbLocations($location)
+                $this->getBreadcrumbLocations($locationId)
             ),
             'columns' => $columns,
         ]);
     }
 
     private function getColumns(
-        Location $location,
+        int $locationId,
         int $limit,
         Query\SortClause $sortClause,
-        bool $gridView = false
+        bool $gridView = false,
+        int $rootLocationId = self::ROOT_LOCATION_ID
     ): array {
-        $breadcrumbLocations = $this->getBreadcrumbLocations($location);
-        $breadcrumbLocationsLast = count($breadcrumbLocations) - 1;
+        $location = $this->locationService->loadLocation($locationId);
+
+        $locationPath = $this->getRelativeLocationPath($rootLocationId, $location->path);
+        $locationPathLast = count($locationPath) - 1;
 
-        $columnLocations = $breadcrumbLocationsLast < 2
-            ? $breadcrumbLocations
+        $columnLocations = $locationPathLast < 4
```

Comment:
It is not. All we need to know here is if `locationPath` has more than 4 elements (because if so we need to pick relevant ones: first one and last three).

But I'll refactor this just a little bit to rely on count here and calculate (-1) last position later.

Reply:
Done